x/pricefeed/simulation/operations/msg.go

```diff
@@ -15,69 +16,53 @@ import (
 )
 
 var (
-	noOpMsg  = simulation.NoOpMsg(pricefeed.ModuleName)
-	btcPrice = sdk.MustNewDecFromStr("7000")
-	bnbPrice = sdk.MustNewDecFromStr("15")
-	xrpPrice = sdk.MustNewDecFromStr("0.25")
+	noOpMsg   = simulation.NoOpMsg(pricefeed.ModuleName)
+	btcPrices = []sdk.Dec{}
+	bnbPrices = []sdk.Dec{}
+	xrpPrices = []sdk.Dec{}
+	genPrices sync.Once
 )
 
-func getStartPrice(marketID string) (startPrice sdk.Dec) {
-	switch marketID {
-	case "btc:usd":
-		return sdk.MustNewDecFromStr("7000")
-	case "bnb:usd":
-		return sdk.MustNewDecFromStr("15")
-	case "xrp:usd":
-		return sdk.MustNewDecFromStr("0.25")
-	}
-	return sdk.MustNewDecFromStr("100")
-}
-
-func getRecentPrice(marketID string) (prevPrice sdk.Dec) {
-	switch marketID {
-	case "btc:usd":
-		return btcPrice
-	case "bnb:usd":
-		return bnbPrice
-	case "xrp:usd":
-		return xrpPrice
-	}
-	return sdk.MustNewDecFromStr("100")
-}
-
-func setRecentPrice(marketID string, newPrice sdk.Dec) {
-	switch marketID {
-	case "btc:usd":
-		btcPrice = newPrice
-		return
-	case "bnb:usd":
-		bnbPrice = newPrice
-		return
-	case "xrp:usd":
-		xrpPrice = newPrice
-		return
-	}
-}
-
-func getIncrement(marketID string) (increment sdk.Dec) {
-	startPrice := getStartPrice(marketID)
-	divisor := sdk.MustNewDecFromStr("20")
-	increment = startPrice.Quo(divisor)
-	return increment
-}
-
 // SimulateMsgUpdatePrices updates the prices of various assets by randomly varying them based on current price
-func SimulateMsgUpdatePrices(keeper keeper.Keeper) simulation.Operation {
+func SimulateMsgUpdatePrices(keeper keeper.Keeper, blocks int) simulation.Operation {
 	// get a pricefeed handler
 	handler := pricefeed.NewHandler(keeper)
 
 	return func(r *rand.Rand, app *baseapp.BaseApp, ctx sdk.Context, accs []simulation.Account) (
 		simulation.OperationMsg, []simulation.FutureOperation, error) {
+		genPrices.Do(func() {
```

Comment:
Good strategy to use `sync.Once` to ensure only run one time and then memoize the aseet prices for fast lookup by `pickNewRandomPrice`